from . import db
from marshmallow import fields, Schema, validate, validates_schema, ValidationError

favorite_cuisines_table = db.Table('favorite_cuisines',
                                   db.Column('user_id', db.Integer,
                                             db.ForeignKey('users.id')),
                                   db.Column('cuisine_id', db.Integer,
                                             db.ForeignKey('cuisines.id')),
                                   db.PrimaryKeyConstraint(
                                       'user_id', 'cuisine_id')
                                   )


class Cuisine(db.Model):
    __tablename__ = 'cuisines'

    id = db.Column(db.Integer, primary_key=True)
    name = db.Column(db.String(128), nullable=False)
    users = db.relationship('User',
                            secondary=favorite_cuisines_table,
                            backref='user_cuisines'
                            )


class User(db.Model):
    __tablename__ = 'users'

    id = db.Column(db.Integer, primary_key=True)
    name = db.Column(db.String(128), nullable=False)
    cuisines = db.relationship('Cuisine',
                               secondary=favorite_cuisines_table,
                               backref='cuisine_users'
                               )
